openedx_user_groups/api.py

```diff
@@ -0,0 +1,418 @@
+            )
+            .annotate(group_count=Count("usergroupmembership__group", distinct=True))
+            .filter(group_count__gt=1)
+        )
+        if duplicates.exists():
+            # TODO: Prompt for action, but for the time being remove the duplicates
+            for duplicate in duplicates:
+                duplicate.usergroupmembership_set.filter(
+                    group__in=group_collection.user_groups.all()
+                ).delete()
+        return group_collection, duplicates
+
+
+def get_available_registered_criteria_schema():
+    """Get all available with their schema for fields, operators and descriptions.
+
+    Returns:
+        dict: A dictionary containing the schema for all available criteria. For example:
+        {
+            "criterion_type": {
+                "fields": {
+                    "field_name": {
+                        "type": "string",
+                        "description": "Description of the field"
+                    }
+                },
+                "operators": ["operator1", "operator2"],
+                "description": "Description of the criterion",
+                "criterion_type": "criterion_type",
+                "supported_scopes": ["course", "organization", "instance"]
+            }
+        }
+    """
+    return {
+        criterion_type: criterion_class.get_schema()
+        for criterion_type, criterion_class in CriterionManager.get_criterion_classes().items()
+    }
+
+
+# TODO: THESE METHODS I HAVEN'T TESTED YET
+
+
+def get_group_by_name_and_scope(name: str, scope: str):
+    """Get a user group by its name and scope.
+
+    Args:
+        name (str): The name of the user group.
+        scope (str): The scope of the user group.
+
+    Returns:
+        UserGroup: The user group.
+    """
+    return UserGroup.objects.get(name=name, scope=scope)
+
+
+def get_user_group_members(group_id: int):
+    """Get the members of a user group.
+
+    Args:
+        group_id (int): The ID of the user group.
+
+    Returns:
+        list: A list of users that are members of the user group.
+    """
+    return UserGroup.objects.get(id=group_id).users.all()
+
+
+def update_group_name_or_description(group_id: int, name: str, description: str):
+    """Update the name or description of a user group.
+
+    Args:
+        group_id (str): The ID of the user group.
+        name (str): The name of the user group.
+        description (str): A brief description of the user group.
+    """
+    UserGroup.objects.filter(id=group_id).update(name=name, description=description)
+
+
+def hard_delete_group(group_id: int):
+    """Hard delete a user group. This will delete the group and all its criteria."""
+    UserGroup.objects.filter(id=group_id).delete()
+
+
+def soft_delete_group(group_id: int):
+    """Soft delete a user group. This will not delete the group, but it will prevent it from being used by disabling it."""
```

Comment:
For some reason I can't comment this on the line below, but I think `is_active` is actually named `enabled`